integrated_channels/blackboard/exporters/content_metadata.py

```diff
@@ -17,7 +17,9 @@ class BlackboardContentMetadataExporter(ContentMetadataExporter):
     DATA_TRANSFORM_MAPPING = {
         'name': 'title',
         'externalId': 'key',
-        'description': 'enrollment_url',
+        'enrollment_metadata': 'enrollment_url',
+        'syllabus_body': 'full_description',
+        'image_url': 'image_url',
```

Comment:
Why not just have a `'content_body': 'content_body'` mapping and then write a `transform_content_body` method that contains all of the values you're using exclusively for the course content page/section here instead of doing a bunch of different popping and copying in the client?

ie the content transform would return: `{'course_content': {'element': 'url', 'link': 'other url', 'image_url': 'value', 'syllabus_body': 'description'}}` and then in the client you could just pop that whole dictionary off and use it.

Even further, you could just do everything in the `generate_integration_content_data` method here instead.

Idk how you guys feel about that, but since the exporter classes all seem to just be doing data transformations, I personally feel like we should just put as many data transformations as we can in the exporter, rather than in the client. There is some weird timing stuff (of the calls and transformations) to consider, but I think what you're adjusting should be fine to do at an exporter level.

Reply:
Yeah, separation of concerns wise, I think it's best, when possible, to do all data work in exporter, so transmitter is just a dumb-ish sender. I found it not possible to avoid breaking this rule when I did the course_id work in Bb as you know, for reasons, but yeah in general I agree with this principle.

Reply:
also makes testing super easy, exporter tests cover all transformations, and transmitter tests don't need to be rigorous then... just a thought on that line too

Reply:
Yeah I also had to do some transformations in Moodle because of its hideous array formatting. I don't expect there to be none there, but think we should _try_ to do them in the exporter and if we find we need something changed because the overall item format needs to be changed, then we can do that in the client.

Reply:
Agreed with the exporter doing the bulk of the work instead of the client. It's difficult to do no work here because we're dealing with three separate posts to create objects on blackboard, all of which require their own formatting. Perhaps there's some consolidation that can be done here. Let me look.

Reply:
You may need that method `generate_integration_content_data` but from what I looked at (and maybe didn't look close enough), you could at the very least group all the data together like my first suggestion. We may not be able to consolidate much more than that and that's okay

Reply:
I've broken content metadata exporting for blackboard up into three sections: 
```
    DATA_TRANSFORM_MAPPING = {
        'course_metadata': 'course_metadata',
        'course_content_metadata': 'course_content_metadata',
        'course_child_content_metadata': 'course_child_content_metadata',
    }
```
Now, we perform 3 transformations that format each of these fields to the exact json that Blackboard is looking for each of these post operations. The only thing that needs to be formatted in the client now is the hashed unique BB id.

Reply:
That is, when making the posts to BB, all we have to do is `content_metadata.get('course_metadata')` or equivelent for each of the steps.